### DISCUSSION TITLE
Connection Error - Failed to connect via rendezvous server: Please try later

### DISCUSSION THREAD

**Discussion by mitchellarganbright** [rustdesk/rustdesk#5817] [Q&A] [answered]
RustDesk Pro server 1.1.10

Happens about 50% of the time, I get this this message when trying to connect Win10 to Win10 or MacOS to Win10. Didn't happen before I upgraded to RustDesk Pro. Both computers have gigabit internet connections. 

![image](https://github.com/rustdesk/rustdesk/assets/13736705/acb548f5-3fc7-4b6c-9b4d-8b44d93a77a3)

**Comment by rustdesk** [comment]
@dinger1986 do you receive similar report from the other Pro users?

**Reply by dinger1986** [reply]
provided all ports are opened it works fine, only one person actually had an issue and they had to define their relay via Web Console - Settings - Relay then all worked fine

**Reply by michele-tn** [reply]
Hi there,


I've been getting the same error for a couple of hours now. I'm using the free Rustdesk servers.

Thank you very much in advance for your feedback.


Greetings,


Michele

**Reply by mitchellarganbright** [reply]
It looks like the issue went away on it's own, I am unable to reproduce the issue today.

**Reply by mitchellarganbright** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/13736705/be516557-2f70-46ec-942d-37017893ea07)
 Looks like I am still having this issue. All of my clients are unreachable.

**Reply by rustdesk** [reply]
There seems some network/server setup issue.

**Reply by rustdesk** [reply]
Not sure if this https://github.com/rustdesk/rustdesk/discussions/5842#discussioncomment-7134239 can help you.

**Reply by mitchellarganbright** [reply]
I'm unsure how it can be a setup issue when it works fine 50% of the time? I will check the relay server, but I have never used that before and never had an issue.

**Reply by rustdesk** [reply]
> 50% of the time

We do not know this either, :(, since you are the first client report this. Anyhow, we will dig our code to find out if there some clue.

**Reply by ronhip** [reply]
I am also experiencing this issue. I cannot connect to any other machine. I've had no issues up until this time. I have re-installed rust desk (Mac Apple Silicon), it also does not work from my phone. Feels like a problem on the rustdesk side. I am not using a self-hosted server.

![Screen Shot 2023-09-28 at 8 02 28 PM](https://github.com/rustdesk/rustdesk/assets/38194019/d34b555c-929f-4369-adda-5eb859570459)

**Reply by rustdesk** [reply]
This is different issue, this is because our public server overloaded.

**Reply by ronhip** [reply]
Cannot ping rustdesk server. Have you tried turning it off and on again?

PING rs-ny.rustdesk.com (155.138.247.159): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4
Request timeout for icmp_seq 5

**Reply by rustdesk** [reply]
Please do not discuss this here, you will distract us from solving real issue.

**Reply by ronhip** [reply]
rustdesk,
Thank you for the VERY quick reply. Any chance you are working to resolve the overload issues? Would it solve my problems to self-host? (Trying to avoid that...)

**Reply by mitchellarganbright** [reply]
Thank you! I’ve also recently noticed that the issue happens much less often when connecting to clients on the same network, whereas clients far away frequently receive this error.

**Reply by mitchellarganbright** [reply]
Any update on what the problem may be? I am still having this issue.

**Reply by dinger1986** [reply]
you have configured your clients correctly? https://rustdesk.com/docs/en/self-host/client-configuration/

**Reply by mitchellarganbright** [reply]
Yes, all clients have the same config and most were deployed with the windows string. 

<img width="897" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/9e42e99f-bb11-4321-86b2-e4e09e2fd9c9">

<img width="603" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/fdf023db-71cf-46ba-86ff-fbf4973edca3">

I see that there is a new rustdesk server update, I will try installing and see if that helps at all.

**Reply by mitchellarganbright** [reply]
Just updated, continuing to get the same error. 

<img width="537" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/6fe497d9-efd4-455b-9e4c-05aebe18dbc0">

**Reply by dinger1986** [reply]
you dont have the api server set so it wont pull the relay server from there

**Reply by dinger1986** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/7244447/419765da-9d07-4ae1-8f0b-ea8b85c63601)
thats my config

**Reply by mitchellarganbright** [reply]
Okay, after adding the API Server I can no longer login to the address book: 
<img width="460" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/231ee9f9-93c1-4288-ae8c-e80b4543fdf1">

**Reply by dinger1986** [reply]
ok, lets go to the start 

You have ports 21114-21117 TCP and 21116 UDP open to your RustDesk Server? 

How did you install your server?

**Reply by mitchellarganbright** [reply]
<img width="785" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/8a793733-68d1-4416-a724-a068af394572">

Yes, here is my port forwarding. 

The previous rustdesk free server was uninstalled, and then the rustdesk pro server was downloaded and installed. This is on windows.

**Reply by dinger1986** [reply]
ok and what do you have in the API field on the client?

Does your router support hairpin nat?

**Reply by mitchellarganbright** [reply]
The API Server is the same as the ID Server, with http:// upfront. 

<img width="519" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/aa7d3176-c0e9-47fe-a6f4-5246cc0c6990">

I am unsure what hairpin nat is, but I am able to access the web console from outside the network. It is a TP-Link ER605

**Reply by mitchellarganbright** [reply]
All of the external domains work internally if that is what you are asking.

**Reply by dinger1986** [reply]
If thats all the ports that have been opened you will need to put it in as http://rustdesk.domain.com:21114 as you havent redirected port 21114 to port 80 (not suggesting you use port redirection, leave it on port 21114)

**Reply by mitchellarganbright** [reply]
Correct, I am able to access the rustdesk web console with that address. Are you saying that my API Server needs to be set to http://rustdesk.<domain>.com:21114 ?

**Reply by dinger1986** [reply]
yes

**Reply by mitchellarganbright** [reply]
Okay, both clients have been reconfigured to use the domain with 21114 port, but I am still getting this message: 

<img width="530" alt="image" src="https://github.com/rustdesk/rustdesk/assets/13736705/9e03e07f-d0b3-4262-8ac0-41f66a2db2c9">

**Reply by mitchellarganbright** [reply]
But I was able to successfully login to the address book this time.

**Reply by dinger1986** [reply, marked as answer]
I was just about to ask that,

Have you setup the relay server in the web console?

So Settings - Relay - + and enter the domain with no ports or anything else, I have it switched on to always use relay for mine, don't know if that right or wrong but works for me

**Reply by dinger1986** [reply]
or you can fill in the relay server on the clients so it matches the ID Server and see if that works

**Reply by mitchellarganbright** [reply]
Brilliant, that has fixed it. Thank you so much for your help.

**Reply by dinger1986** [reply]
no problems, out of interest what route did you go down? if its set in the web console its easier to deploy out to other clients

**Reply by mitchellarganbright** [reply]
I put the relay server into the web console. It appears that the API server does not need to be specified, as I have removed it from the client and it is still functioning perfectly.

**Reply by dinger1986** [reply]
ok cool, the API server should need to be specified if logging in, good to know though :) enjoy!

**Reply by mitchellarganbright** [reply]
I am still able to login without the api server specified :)

**Reply by dinger1986** [reply]
there you go!

**Reply by mitchellarganbright** [reply]
You may need to specify the API server if you use port 80, because it might autofill with port 21114.

**Reply by pdifeo** [reply]
> and enter the domain with no ports or anything else

@dinger1986 I see that you also run into the same mistake as me. @Mouxy  says you have to use the ip address and not the domain name

https://github.com/rustdesk/rustdesk-server-pro/issues/102#issuecomment-1745344975

**Reply by dinger1986** [reply]
Works absolutely fine for me using the dns name, you should be able to use the IP or domain just not adding the port

**Reply by pdifeo** [reply]
Please read the whole linked thread. If the relay server is turned off, the "health check" does not work. If I understand you and @Mouxy are from the RustDesk team, you need to talk to each other.

**Reply by dinger1986** [reply]
I did, I have no idea who Mouxy is, until you linked to that thread I had never seen the name before, the dns name works absolutely fine for purposes of using with one relay, I know because I have been using it like that since pro came out over circa 1000 devices and never have issues like listened in that thread, there is no relevance to that thread here.

@rustdesk please lock this thread

**Reply by Mouxy** [reply]
When deploying multiple hbbr servers to use for GeoLocation, the dns names wouldn't work, we tested it several times, provided logs, etc

I'm a paying customer that decided to help you writing the documentation, but I just realised how pissed off you are now because someone that paid for your product decided to use his time to help you write your docs.

**Reply by pdifeo** [reply]
I am sorry to have triggered this discussion. I didn't mean to exacerbate it.

**Reply by rustdesk** [reply]
@Mouxy, we are very sorry for disturbing you.

**Reply by dinger1986** [reply]
Sorry, I'm not pissed at all (also not helping me as I am just doing my best to help out on a volunteer basis not in any commercial capacity).

So for my own understanding for more than 1 it needs an IP rather than DNS? Is that what you found? As for me it works fine with DNS with only one but I don't have multiple defined similar as the original OP hence my confusion of the other issue being linked

**Comment by michele-tn** [comment]
https://github.com/rustdesk/rustdesk-server/issues/311#issuecomment-1735537137


It seems that now the situation has been restored, I managed to connect.

Thanks so much for the feedback.

Greetings,

Michele

**Comment by norrislebron-beep** [comment]
io ho la stessa problematica con l'app ios e pc win 11 ma non ho un server selfhosted
